src/chains.rs

```diff
@@ -74,6 +74,60 @@ use crate::utils::{
     rewrite_ident, trimmed_last_line_width, wrap_str,
 };
 
+/// Provides the original input contents from the span
+/// of a chain element with trailing spaces trimmed.
+fn format_overflow_style(span: Span, context: &RewriteContext<'_>) -> Option<String> {
+    context.snippet_provider.span_to_snippet(span).map(|s| {
+        s.lines()
+            .map(|l| l.trim_end())
```

Comment:
This is a topic that is probably best deferred to a subsequent PR once this is actually a reachable codepath, but the thinking is essentially that if we're going to have variants for the to-be-added config option which utilizes the simple/basic overflow strategy of just spitting back out the element contents when that element won't fit, then we can probably still give the user an assist and trim any trailing whitespace characters off those lines.

Reply:
Thanks, that was a helpful explanation!